EducationPortal.DAL/FileStorage.Core/Infrastructure/Storage.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using EducationPortal.DAL.Entities;
+using EducationPortal.DAL.FileStorage.Core.Infrastructure.Interfaces;
+
+namespace EducationPortal.DAL.FileStorage.Core.Infrastructure
+{
+    public class Storage : IDatabase
+    {
+        private readonly FSContext context;
+        private readonly string name;
+
+        public Storage(FSContext context)
+        {
+            this.context = context;
+            this.name = typeof(Storage).Namespace;
+            // this.name = Assembly.GetExecutingAssembly().FullName;
+
+            // this.EnsureCreated();
+        }
+
+        // true if created, false if it already existed
+        public virtual bool EnsureCreated()
+        {
+            if (!this.Exists())
+            {
+                try
+                {
+                    this.Create();
+                    this.CreateTables();
+                    return true;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine($"Failed to create storage. Message: {e.Message} " +
+                                      $"In method: {e.TargetSite} StackTrace: {e.StackTrace}");
+                    throw;
+                }
+            }
+
+            return false;
+        }
+
+        // Create storage(directory)
+        public void Create()
+        {
+            Directory.CreateDirectory(this.name);
```

Comment:
Will it create a directory with a namespace name?
this.name = typeof(Storage).Namespace;

Reply:
Namespace is a default name. Updated to specified storage name if it defined.
if (!string.IsNullOrEmpty(storageName))
            {
                this.name = storageName;
            }
            else
            {
                this.name = typeof(Storage).Namespace;
            }